sw.js

```diff
@@ -0,0 +1,50 @@
+const cacheName = 'v1';
+
+const cacheAssets = [
+  'blog.html',
+  'example.html',
+  'faq.html',
+  'handbook.html',
+  'index.html',
+  'mutators.html',
+  'plugins.html',
+  'quickstart.html',
+  '/dist/all.css',
+  '/dist/all.bundle.js'
+];
+
+// Call Install Event
+self.addEventListener('install', e => {
+  console.log('Service Worker: Installed');
+
+  e.waitUntil(
+    caches.open(cacheName).then(cache => {
+        console.log('Service Worker: Caching Files');
+        cache.addAll(cacheAssets);
+      }).then(() => self.skipWaiting())
+  );
+});
+
+// Call Activate Event
+self.addEventListener('activate', e => {
+  console.log('Service Worker: Activated');
+  // Remove unwanted caches
+  e.waitUntil(
+    caches.keys().then(cacheNames => {
+      return Promise.all(
+        cacheNames.map(cache => {
+          if (cache !== cacheName) {
+            console.log('Service Worker: Clearing Old Cache');
+            return caches.delete(cache);
+          }
+        })
+      );
+    })
+  );
+});
+
+// Call Fetch Event
+self.addEventListener('fetch', e => {
+  console.log('Service Worker: Fetching');
+  e.respondWith(fetch(e.request).catch(() => caches.match(e.request)));
```

Comment:
Doesn't this first try to fetch, and then attempt to get from cache if the request fails. Wouldn't we want it the other way around, so the cache is called first?

Reply:
well @hugo-vrijswijk, you may have a point. Now it works like -> check if there is connection -> if yes, load using web, if not try get cache.